cityofaustin/atd-vz-data
# Create separate Hasura permission group for VZE Admins

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/11893

This PR updates the VZE stack to use a new admin role called `vz-admin` instead of the default Hasura admin. The admin permissions are now much more restricted since only a few actions require more permissions than the existing `editor` role. Almost all of the new permissions are identical to the `editor` role, and you can see the exceptions [here](https://github.com/cityofaustin/atd-data-tech/issues/11893#issuecomment-1965205583).

****PLEASE USE THE HASURA CONSOLE TO INSPECT THESE PERMISSION CHANGES.**** Unless you really wanna look at them, it is hard to read through the metadata diffs! Also, I flagged the other code updates to review with comments.

[Here is the launch plan](https://github.com/cityofaustin/atd-data-tech/issues/11893#issue-1648187425)

## Testing
**URL to test:** 
Local

**Steps to test:**
1. Start the local stack and apply the updated metadata with `hasura metadata apply`
2. If you are already signed in to the local VZE app, sign out and sign in using [this VZ admin test account](https://start.1password.com/open/i?a=RPNXYTT5QNABRL3IE5XBV3362I&v=toa6yd4fjoyobe2x57tqkpcz5e&i=je2hwb2um3pd3tqdawukr33vzi&h=austintransportation.1password.com) which uses the new `vz-admin` role and not the current `admin` role.
3. Go to http://localhost:3000/#/profile, and you should see `vz-admin` listed in the allowed roles array.
4. Go to the Fatalities view http://localhost:3000/#/fatalities, make sure it loads, filters, and exports correctly, and then go to a crash listed in the table
5. Edit every field on the crash field. The parts that need admin permissions are:
   - Edit the name of the person who passed away. If needed, update a person and primary person record to fatality to test the name update on both tables.
   - Fatality Review Board Recommendations section (update each field)
   - Notes section (add, update, and delete)
6. Go to the Locations view http://localhost:3000/#/locations and make sure it loads, filters, and exports
7. Go to a location page and test the global export and CR3 and non-CR3 table exports. Add, edit, and remove a location note.
8. Make sure the Inconsistent KSI Counts view loads http://localhost:3000/#/reports/inconsistent_ksi_counts. This is another admin-specific permission needed. Searching by Crash ID is broken in prod and this branch, but I think that we've decided that the VZ team doesn't use this view anymore.
9. Test the non-CR3 upload tool http://localhost:3000/#/tools/upload_non_cr3. I used the example template and removed all of the rows except for the first one to upload into the local DB.
10. Test creating a temporary crash record with a unit. http://localhost:3000/#/tools/create_crash_record
11. Test the Users view http://localhost:3000/#/users. It should not load since the VZ API updates in this PR are not deployed to the staging API. I have run the API locally to confirm that the new role works. If you want to test this locally:
    - Update the `.env` in the `atd-vze` folder to point to `REACT_APP_CR3_API_DOMAIN="http://localhost:3010"` and restart the React app
    - Follow [the steps in the readme](https://github.com/cityofaustin/atd-vz-data/tree/master/atd-cr3-api#running-the-example) to start it up the API server locally. You will then be able to fetch users data for the user list and user pages.

---
#### Ship list
- [ ] Make one last edit through the Hasura console to get Hasura formatting changes into code before merge
